product_cost_price_avco_sync/models/stock_move.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.tools.float_utils import float_compare
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def write(self, vals):
+        res = super(StockMove, self).write(vals)
+        if ('price_unit' in vals and
+                not self.env.context.get('skip_cost_update')):
+            self.price_unit_update()
+        return res
+
+    def price_unit_update(self):
+        procesed_moves = []
+        for move in self.sorted('date'):
+            if (move.id in procesed_moves or move.state != 'done' or
+                    not move._is_in() or
+                    move.product_id.cost_method != 'average'):
+                continue
+            # not take in account move's quantities
+            previous_qty = move.product_id.with_context(
+                to_date=move.date
+            ).qty_available - move.quantity_done
+            price_history = self.product_id.get_cost_history_date(move.date)
+            price = move.get_average_price(previous_qty, price_history)
+            previous_qty += move.quantity_done
+            affected_moves = self.search([
+                ('product_id', '=', move.product_id.id),
+                ('date', '>', move.date),
+            ], order='date')
+            for af_move in affected_moves:
+                if af_move._is_in():
+                    if af_move.inventory_id:
+                        af_move.with_context(
+                            skip_cost_update=True).price_unit = price
+                    # Avoid reprocess move if included in self.write
+                    elif af_move.id in procesed_moves:
+                        procesed_moves.append(af_move.id)
+                    price = af_move.get_average_price(previous_qty, price)
+                    previous_qty += af_move.quantity_done
+                elif af_move._is_out():
+                    af_move.with_context(
+                        skip_cost_update=True).price_unit = -price
+                    previous_qty -= af_move.quantity_done
+                    if previous_qty < 0.0:
```

Comment:
Why this adjustment? Standard is not considering this AFAIK and the idea is to recreate the same computation.

Reply:
Answer not given

Reply:
Answer not given